src/tasks/task_list.hpp

```diff
@@ -93,6 +93,17 @@ class TaskList {
     return id;
   }
 
+  // overload to add member functions of class T to task list
+  // NOTE: we must capture the object pointer
+  template <class F, class T, class... Args>
+  TaskID AddTask(F func, T *obj, TaskID &dep, Args &&... args) {
```

Comment:
This is a bit nitpicking on my part, but I would recommend moving the argument T *obj over, so that the function prototype is more consistent with the previous AddTask call.

Reply:
this signature was motivated by consistency with things like `std::async` calls with member functions.